### ISSUE TITLE
Standalone application has very crackly audio when set to non-44.1khz sample rate.

### ISSUE THREAD

**Issue by a2aaron** [robbert-vdh/nih-plug#60]
I am on an M1 Apple Macbook Pro using CoreAudio running macOS Monterey 12.6. When running a plugin via the standalone binary, crackly audio occurs unless the sample rate is set to 44100. Note that this doesn't seem to be an issue with the plugin, but rather with how the standalone binary uses the CoreAudio backend.

To reproduce, compile the [`poly_mod_synth`](https://github.com/robbert-vdh/nih-plug/tree/master/plugins/examples/poly_mod_synth) example as a standalone binary. Note that I had to modify the plugin's existing `Cargo.toml` and add a `main.rs` file since it was not already configured to make a standalone binary (see below for exact changes I made)

```rust
[package]
name = "poly_mod_synth"
version = "0.1.0"
edition = "2021"
authors = ["Robbert van der Helm <mail@robbertvanderhelm.nl>"]
license = "ISC"

[lib]
crate-type = ["cdylib", "lib"] # adding "lib" to the crate type

[dependencies]
nih_plug = { path = "../../../", features = ["assert_process_allocs", "standalone"] } # adding "standalone" feature

rand = "0.8.5"
rand_pcg = "0.3.1"
```

```rust
// poly_mod_synth/main.rs
use poly_mod_synth::PolyModSynth;

fn main() {
    nih_plug::nih_export_standalone::<PolyModSynth>();
}

// additionally, mark PolyModSynth as public in lib.rs
```

When running the plugin using `./poly_mod_synth --midi-input "USB Axiom 49 Port 1"`, there will be crackly audio. When running with `./poly_mod_synth --midi-input "USB Axiom 49 Port 1" --sample-rate 44100`, no crackly audio is heard.

Additionally, when I run the synth as a vst3 plugin and play it in vPlayer 3, I do not hear crackly audio regardless of the sample rate.

I have not tested the plugin on the JACK audio backend (mostly because I do not have that set up).

**Comment by robbert-vdh** [comment]
Good catch on https://github.com/RustAudio/cpal/issues/748. I don't use macOS but I also noticed the same heavily distorted sound when I was making sure things built correctly on a mid-2012 13" MBP running Catalina. I looked into it for a bit because I assumed I must have been doing something shady with unsafe lifetime management but that wasn't the case, and I forgot about this after that. Sometimes standalone audio on macOS works just fine, other times it's distorted. I don't know much about CoreAudio, but it may depend on what other applications are doing. Some modern audio APIs like PipeWire allow applications to use different period sizes and sampel rates, but CoreAudio might not.
